core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -524,7 +547,7 @@ private static String httpHeaderValue(Parameter entry, HttpRequest req) {
             } else if (clazz == String.class) {
                 return (T) str;
             }
-        } catch (NumberFormatException e) {
+        } catch (IllegalArgumentException e) {
```

Comment:
Since you take out the enum check here, you can change it back.

Reply:
Checked & Updated Thanks